maat-court-data-api/src/test/java/gov/uk/courtdata/integration/repOrder/RepOrderControllerIntegrationTest.java

```diff
@@ -310,4 +320,111 @@ void givenValidRepId_whenFindIOJAssessorDetailsIsCalled_() throws Exception {
+                .andExpect(jsonPath("$.length()").value(2))
+                .andExpect(jsonPath("$", Matchers.containsInAnyOrder(repOrderValid.getId(), repOrderValid2.getId())));
+    }
+
+    @Test
+    void givenSingleAsk_whenFdcDelayedCalled_thenOnlyOneValidRepOrdersReturned() throws Exception {
+        setUpFdcMinDelayAppliesEntities();
+        mockMvc.perform(MockMvcRequestBuilders.get(buildFdcDelayedEndpointURL(5,LocalDate.now(), 1))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andExpect(jsonPath("$", Matchers.containsInAnyOrder(repOrderValid.getId())));
+    }
+
+    @Test
+    void givenTooLargeAsk_whenFdcFastTrackCalled_thenAllAvailableValidRepOrdersReturned() throws Exception {
+        setUpFdcMinDelayAppliesEntities();
+        mockMvc.perform(MockMvcRequestBuilders.get(buildFdcFastTrackEndpointURL(5, LocalDate.now(), 5))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(2))
+                .andExpect(jsonPath("$", Matchers.containsInAnyOrder(repOrderFuture.getId(), repOrderFuture2.getId())));
+    }
+
+    @Test
+    void givenSingleAsk_whenFdcFastTrackCalled_thenOnlyOneValidRepOrdersReturned() throws Exception {
+        setUpFdcMinDelayAppliesEntities();
+        mockMvc.perform(MockMvcRequestBuilders.get(buildFdcFastTrackEndpointURL(5, LocalDate.now(), 1))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andExpect(jsonPath("$", Matchers.containsInAnyOrder(repOrderFuture.getId())));
+    }
+    private String buildFdcDelayedEndpointURL(int delayPeriod, LocalDate dateReceived, int numRecords){
+        //delay=5&dateReceived=01.10.2010&numRecords=5
+        String date = dateReceived.format(DateTimeFormatter.ISO_DATE);
+        String parameterString = "?delay=%s&dateReceived=%s&numRecords=%s".formatted(delayPeriod, date, numRecords);
+        return FDC_DELAYED_ENDPOINT_URL +parameterString;
+    }
+
+    private String buildFdcFastTrackEndpointURL(int delayPeriod, LocalDate dateReceived, int numRecords){
+        //delay=5&dateReceived=01.10.2010&numRecords=5&startingMonth=0
+        String date = dateReceived.format(DateTimeFormatter.ISO_DATE);
+        String parameterString = "?delay=%s&dateReceived=%s&numRecords=%s".formatted(delayPeriod, date, numRecords);
+        return FDC_FAST_TRACK_ENDPOINT_URL +parameterString;
```

Comment:
Rather than use String format I would consider using the built in functionality of the MVC test framework and use placeholders
e.g.
`mockMvc.perform(MockMvcRequestBuilders.get(FDC_DELAYED_ENDPOINT_URL, 5, date, 5)
`where
`    private static final String FDC_DELAYED_ENDPOINT_URL = "/api/internal/v1/assessment/rep-orders/eligible-for-fdc-delayed-pickup?delay={delay}&dateReceived={dateReceived}&numRecords={numRecords}";`